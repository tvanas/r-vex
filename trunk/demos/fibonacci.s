add $r0.15 = $r0.0, 1
mov $r0.1 = $r0.0
add $r0.10 = $r0.0, 44
add $r0.2 = $r0.0, 1
;;
LABEL_BEGIN:
add $r0.2 = $r0.1, $r0.2
add $r0.3 = $r0.0, $r0.2
cmpeq $b0.0 = $r0.9, $r0.10
br $b0.0, LABEL_END
;;
add $r0.9 = $r0.9, 1
add $r0.1 = $r0.0, $r0.3
goto LABEL_BEGIN
;;
LABEL_END:
stw 0x0[$r0.15] = $r0.1
mov $r0.9 = $r0.0
;;
