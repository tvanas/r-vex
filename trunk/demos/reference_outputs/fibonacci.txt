r-VEX
-----
Cycles: 0x00000231

Data memory dump

addr | contents
-----+-----------
0x00 | 0x43A53F82
0x01 | 0x00000000
0x02 | 0x00000000
0x03 | 0x00000000
0x04 | 0x00000000
0x05 | 0x00000000
0x06 | 0x00000000
0x07 | 0x00000000
0x08 | 0x00000000
0x09 | 0x00000000
0x0A | 0x00000000
0x0B | 0x00000000
0x0C | 0x00000000
0x0D | 0x00000000
0x0E | 0x00000000
0x0F | 0x00000000
